src/lib/core/util.c

```diff
@@ -484,3 +484,16 @@ thread_sleep(double sec)
 	assert(rc == 0);
 	(void)rc;
 }
+
+char *
+getenv_limited(const char *name, size_t buf_size, bool *exceeded)
```

Comment:
Can you add a unit test for this function?

Reply:
Done:
<details><summary>diff</summary>

```diff
diff --git a/test/unit/CMakeLists.txt b/test/unit/CMakeLists.txt
index 1f29ee91c..957eeb153 100644
--- a/test/unit/CMakeLists.txt
+++ b/test/unit/CMakeLists.txt
@@ -563,3 +563,8 @@ create_unit_test(PREFIX key_def
                  SOURCES key_def.c box_test_utils.c
                  LIBRARIES unit box core
 )
+
+create_unit_test(PREFIX getenv_safe
+                 SOURCES getenv_safe.c core_test_utils.c
+                 LIBRARIES unit core
+)
diff --git a/test/unit/getenv_safe.c b/test/unit/getenv_safe.c
new file mode 100644
index 000000000..cdbc03baf
--- /dev/null
+++ b/test/unit/getenv_safe.c
@@ -0,0 +1,51 @@
+#include <stdlib.h>
+#include <string.h>
+#include "trivia/util.h"
+
+#define UNIT_TAP_COMPATIBLE 1
+#include "unit.h"
+
+static void
+test_getenv_safe(void)
+{
+	header();
+	plan(10);
+
+	static const char str[] = "some env value";
+	static const char name[] = "TT_GH_7797_ENV_TEST";
+	size_t size = sizeof(str);
+	bool exceeded = true;
+
+	is(getenv(name), NULL, "Getenv finds nothing initially");
+	is(getenv_safe(name, size, &exceeded), NULL,
+	   "Getenv_safe finds nothing");
+	is(exceeded, false, "Exceeded holds false when nothing found");
+
+	is(setenv(name, str, 1), 0, "Setenv succeeeds");
+
+	char *ret1 = getenv(name);
+	isnt(ret1, NULL, "Getenv finds the value");
+	char *ret2 = getenv_safe(name, size, &exceeded);
+	isnt(ret2, NULL, "Getenv_safe finds the value");
+	is(strcmp(ret1, ret2), 0, "Returns are the same");
+	is(exceeded, false, "Exceeded is false when value fits in size");
+	char *ret3 = getenv_safe(name, size - 1, &exceeded);
+	is(ret3, NULL, "Getenv_safe returns nothing when size doesn't fit");
+	is(exceeded, true, "Exceeded is true when size doesn't fit");
+
+	unsetenv(name);
+
+	footer();
+}
+
+int
+main(void)
+{
+	header();
+	plan(1);
+
+	test_getenv_safe();
+
+	footer();
+	return check_plan();
+}
```
</details>